subsys/net/ip/tcp.c

```diff
@@ -1962,6 +1962,21 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		goto next_state;
 	}
 
+	if (th && (conn->state != TCP_LISTEN) && (conn->state != TCP_SYN_SENT) &&
+	    tcp_validate_seq(conn, th) && FL(&fl, &, SYN)) {
+		/* According to RFC 793, ch 3.9 Event Processing, receiving SYN
+		 * once the connection has been established is an error
+		 * condition, reset should be sent and connection closed.
+		 */
+		NET_ERR("conn: %p, SYN received in %s state, dropping connection",
```

Comment:
Are you sure we want to flag this as an error, after all it is not an error in our code but an error in a packet we receive. This could cause huge load in our system if we receive big number of these (denial of service). I would suggest a normal debug message here.

Reply:
Good point, I've lowered the log level to debug.